src/suites/cts/textures/zero_init.spec.ts

```diff
@@ -0,0 +1,1244 @@
+
+        for (let level = mipRange.base; level < mipRange.base + mipRange.count; ++level) {
+          for (let layer = sliceRange.base; layer < sliceRange.base + sliceRange.count; ++layer) {
+            assert(this.params.dimension == '2d');
+
+            const width = this.textureWidth >> level;
+            const height = this.textureHeight >> level;
+
+            const bytesPerPixel = kTextureFormatInfo[this.params.format].bytes
+            const minBytesPerRow = width * bytesPerPixel;
+            const bytesPerRow = Math.ceil(minBytesPerRow / 256) * 256;
+
+            const byteLength = bytesPerRow * height;
+            const buffer = this.device.createBuffer({
+              size: byteLength,
+              usage: C.BufferUsage.CopySrc | C.BufferUsage.CopyDst,
+            });
+
+            const commandEncoder = this.device.createCommandEncoder();
+            commandEncoder.copyTextureToBuffer(
+              { texture, mipLevel: level, arrayLayer: layer },
+              {
+                buffer, bytesPerRow, rowsPerImage: height,
+                // @ts-ignore
+                rowPitch: bytesPerRow, imageHeight: height,
+              },
+              { width, height, depth: 1 }
+            );
+
+            this.device.defaultQueue.submit([commandEncoder.finish()]);
+
+            const expectedPixelData = new Uint8Array(this.gpuDataRepresentation[state]);
+            assert(expectedPixelData.byteLength === bytesPerPixel);
+
+            const arrayBuffer = new ArrayBuffer(byteLength);
+            for (let row = 0; row < height; ++row) {
+              for (let byteOffset = 0; byteOffset < minBytesPerRow; byteOffset += bytesPerPixel) {
+                const pixel = new Uint8Array(arrayBuffer, row * bytesPerRow + byteOffset, bytesPerPixel);
+                pixel.set(expectedPixelData);
+              }
+            }
+
+            this.expectContents(buffer, new Uint8Array(arrayBuffer));
+          }
+        }
+
+      } break;
+
+      case ReadMethod.CopyToTexture: {
+        const { mipRange, sliceRange } = subresourceRange;
+
+        for (let level = mipRange.base; level < mipRange.base + mipRange.count; ++level) {
+          for (let layer = sliceRange.base; layer < sliceRange.base + sliceRange.count; ++layer) {
+            assert(this.params.dimension == '2d');
+
+            const width = this.textureWidth >> level;
+            const height = this.textureHeight >> level;
+
+            const dst = this.device.createTexture({
+              size: [width, height, 1],
+              format: this.params.format,
+              usage: C.TextureUsage.CopyDst | C.TextureUsage.CopySrc,
+            });
+
+            const commandEncoder = this.device.createCommandEncoder();
+            commandEncoder.copyTextureToTexture(
+              { texture, mipLevel: level, arrayLayer: layer },
+              { texture: dst, mipLevel: 0, arrayLayer: 0 },
+              { width, height, depth: 1 },
+            );
```

Comment:
Didn't actually read this code, but after this point shouldn't this be able to share code with the CopyToBuffer case?